lark/load_grammar.py

```diff
@@ -648,17 +649,67 @@ def compile(self, start):
         return terminals, compiled_rules, self.ignore
 
 
+class FromPackageLoader(object):
+    """
+    Provides a simple way of creating custom import loaders that load from packages via ``pkgutil.get_data`` instead of using `open`.
+    This allows them to be compatible even from within zip files.
+    
+    Relative imports are handled, so you can just freely use them.
+    
+    pkg_name: The name of the package. You can probably provide `__name__` most of the time
+    search_paths: All the path that will be search on absolute imports.
+    """
+    def __init__(self, pkg_name, search_paths=("", )):
+        self.pkg_name = pkg_name
+        self.search_paths = search_paths
+        
+    def __repr__(self):
+        return "%s(%r, %r)" % (type(self).__name__, self.pkg_name, self.search_paths)
+
+    def __call__(self, base_paths, grammar_path):
+        if len(base_paths) == 0:
+            to_try = self.search_paths
+        else:
+            assert len(base_paths) == 1
```

Comment:
Why is `len(base_paths)` only 0 or 1? Maybe it should be `Optional[str]` instead then?

Reply:
This is the design you choose in the implementation. I have nothing against changing this, but I imagine you at some point in the future might want to add additional elements to `base_paths`. If that is the case, this `assert` will trigger and we have to change something here. I could future proof this now if you want. (e.g. allow multiple `base_paths` entries.

Reply:
When I used `base_paths`, I did not differentiate between base_paths and search_paths, and I did not consider relative imports vs absolute imports.

Now that we make the distinction, there shouldn't be more than one base_path (or so it seems to me). And search_paths has now taken its place for a list of paths. So I think it should be turned into an optional string.